testr/testr.go

```diff
@@ -59,6 +75,20 @@ func NewWithOptions(t *testing.T, opts Options) logr.Logger {
 	return logr.New(l)
 }
 
+// NewFromInterfaceWithOptions returns a logr.Logger that prints through a
+// TestingT object.
+// In contrast to the simpler New, output formatting can be configured.
+func NewFromInterfaceWithOptions(t TestingT, opts Options) logr.Logger {
```

Comment:
I think a single `NewWithInterface` may be okay. It matches the `NewWithOptions` pattern.

We only need a single variant of it. That we have `New` and `NewWithOptions` is because of historic reasons, we don't need to repeat that.

Reply:
I see, thanks for the explanation.
I renamed the variant with the `Options` parameter `NewWithInterface` and removed the other constructor.